File: ulohy/24-posun/reseni/main.py
def read_int() -> int:
    return int(input())


if __name__ == "__main__":
    s = read_int()
    n = read_int()
    input()

    numbers = []

    for i in range(n):
        numbers.append(read_int())

    if n > 0:
        s %= n

    numbers_shifted = [0] * n
    for i in range(n):
        new_index = i + s
        if new_index < 0:
            new_index += n
        elif new_index >= n:
            new_index -= n

        numbers_shifted[new_index] = numbers[i]

    for i in range(n):
        print(numbers_shifted[i])

File: ulohy/24-posun/reseni/main.cs
using System;

internal class Program
{
    static void Main(string[] args)
    {
        int s = ReadInt();
        int n = ReadInt();
        Console.ReadLine();

        if (n > 0)
        {
            // Neboli `s = s % n`.
            s %= n;
        }

        int[] numbers = new int[n];

        for (int i = 0; i < n; i++)
        {
            numbers[i] = ReadInt();
        }

        int[] numbersShifted = new int[n];
        for (int i = 0; i < n; i++)
        {
            int newIndex = i + s;
            if (newIndex < 0)
            {
                newIndex += n;
            }
            else if (newIndex >= n)
            {
                newIndex -= n;
            }

            numbersShifted[newIndex] = numbers[i];
        }

        for (int i = 0; i < n; i++)
        {
            Console.WriteLine(numbersShifted[i]);
        }
    }

    static int ReadInt()
    {
        return int.Parse(Console.ReadLine());
    }
}
